x-pack/auditbeat/processors/add_session_metadata/add_session_metadata_test.go

```diff
@@ -0,0 +1,211 @@
+						Ppid: uint32(50),
+						Pgid: uint32(100),
+						Sid: uint32(40),
+					},
+					Cwd: "/",
+					Filename: "/bin/ls",
+				},
+				{
+					Pids: types.PidInfo{
+						Tid: uint32(50),
+						Tgid: uint32(50),
+						Ppid: uint32(40),
+						Sid: uint32(40),
+					},
+				},
+				{
+					Pids: types.PidInfo{
+						Tid: uint32(40),
+						Tgid: uint32(40),
+						Ppid: uint32(1),
+						Sid: uint32(1),
+					},
+				},
+			},
+			input: beat.Event{
+				Fields: mapstr.M{
+					"process": mapstr.M{
+						"pid": uint32(100),
+					},
+				},
+			},
+			expected: beat.Event{
+				Fields: mapstr.M{
+					"process": mapstr.M{
+						"executable": "/bin/ls",
+						"working_directory": "/",
+						"pid": uint32(100),
+						"parent": mapstr.M{
+							"pid": uint32(50),
+						},
+						"session_leader": mapstr.M{
+							"pid": uint32(40),
+						},
+						"group_leader": mapstr.M{
+							"pid": uint32(100),
+						},
+					},
+				},
+			},
+			expect_error: false,
+		},
+		{
+			testName: "No Pid Field in Event",
+			config: Config{
+				ReplaceFields: false,
+				PidField: "process.pid",
+			},
+			input: beat.Event{
+				Fields: mapstr.M{
+					"process": mapstr.M{
+						"executable": "ls",
+						"working_directory": "/",
+						"parent": mapstr.M{
+							"pid": uint32(100),
+						},
+					},
+				},
+			},
+			expect_error: true,
+		},
+		{
+			testName: "Pid Not Number",
+			config: Config{
+				ReplaceFields: false,
+				PidField: "process.pid",
+			},
+			input: beat.Event{
+				Fields: mapstr.M{
+					"process": mapstr.M{
+						"pid": "xyz",
+						"executable": "ls",
+						"working_directory": "/",
+						"parent": mapstr.M{
+							"pid": uint32(50),
+						},
+					},
+				},
+			},
+			expect_error: true,
+		},
+		{
+			testName: "PID not in DB",
+			config: Config{
+				ReplaceFields: false,
+				PidField: "process.pid",
+			},
+			input: beat.Event{
+				Fields: mapstr.M{
+					"process": mapstr.M{
+						"pid": "100",
+						"executable": "ls",
+						"working_directory": "/",
+						"parent": mapstr.M{
+							"pid": uint32(100),
+						},
+					},
+				},
+			},
+			expect_error: true,
+		},
+	}
+
+	logger = logp.NewLogger("add_session_metadata_test")
+)
+
+func TestEnrich(t *testing.T) {
+	for _, tt := range enrichTests {
+		reader := procfs.NewMockReader()
+		db := processdb.NewSimpleDB(reader, *logger)
+
+		for _, ev := range tt.mockProcesses {
+			err := db.InsertExec(ev)
+			assert.NoError(t, err, "%s: inserting exec to db: %w", tt.testName, err)
+		}
+		s := addSessionMetadata{
+			logger: logger,
+			db: db,
+			config: tt.config,
+		}
+
+		actual, err := s.enrich(&tt.input)
+		if tt.expect_error {
+			assert.Error(t, err, "%s: error unexpectedly nil", tt.testName)
+		} else {
+			assert.Nil(t, err, "%s: enrich error: %w", tt.testName, err)
+			assert.NotNil(t, actual, "%s: returned nil event", tt.testName)
+			//Validate output
+			eq, msg := compareMapstr(tt.expected.Fields, actual.Fields)
```

Comment:
There are good libs to support this kind of test comparison. I think testify also supports it.

Reply:
I've changed to use `go-cmp` library for this